R/dv8.R

```diff
@@ -0,0 +1,254 @@
+#' Creates a gitlog numstat file from a git file
+#'
+#' From a git file *.git, a gitlog numstat file *.txt is created
+#'
+#' @param git_repo_path path to local repository
+#' @param git_numstat_path file name of git log output
+#'
+#' @return path to git log output
+#' @export
+#' @seealso \code{\link{dv8_gitnumstat_to_hdsmb}} for creating hsdsmb binary file from a git numstat text file
+#'
+dv8_gitlog_to_gitnumstat <- function(git_repo_path, git_numstat_path) {
+  system2(command = "git",
+          args = c("--git-dir",
+                   git_repo_path,
+                   "log",
+                   "--numstat",
+                   "--date=iso"),
+          stdout = git_numstat_path
+          )
+  return(git_numstat_path)
+}
+
+
+#' Converts a git log text file into a hdsmb binary file
+#'
+#' An input file *.txt is converted to *.dv8-dsm
+#'
+#' @param dv8_path path to DV8 binary
+#' @param git_numstat_path path to local git log text file obtained by \code{\link{dv8_gitlog_to_gitnumstat}}
+#' @param start start of date/time range in ISO format ie. 2017-07-08T00:00:00Z
+#' @param stop end of date/time range in ISO format ie. 2018-07-08T00:00:00Z
+#' @param max_cochange_count maximum count of co-changed files per commit, default 1000
+#' @param hdsmb_path name of output hdsmb binary file, defaults to name of input
+#' @param params_output_file output file used to record parameters
+#'
+#' @return path to hdsmb binary file
+#' @export
+#' @seealso \code{\link{dv8_hsdsmb_to_decoupling_level}} to calculate decoupling level from hsdsmb binary file
+#' @seealso \code{\link{dv8_hsdsmb_to_hierclsxb}} to compute design rule hierarchy from hsdsmb binary file
+#' @seealso \code{\link{dv8_hsdsmb_drhier_to_excel}} to export a hsdsmb binary file as an excel file
+#' @seealso \code{\link{dv8_hsdsmb_dsmb_to_dsmj}} to export a hsdsmb binary file as a json file
+#'
+dv8_gitnumstat_to_hdsmb <- function(dv8_path,
```

Comment:
This function is working, but returning errors when executed from R Studio code block (not knit): 

Show in New Window
ERROR: Unable to open matrix file for writing '/tmp/helix-hdsm.dv8-dsm'
[1] "/tmp/helix-hdsm.dv8-dsm"

Reply:
You can disregard this one @nicoelee123 

Apparently, dv8 throws this if you attempt to use it on a folder without writing permissions (in which case was /tmp/ for me).